packages/explat-client/src/create-explat-client.ts

```diff
@@ -162,31 +162,50 @@ export function createExPlatClient( config: Config ): ExPlatClient {
 			}
 		},
 		dangerouslyGetExperimentAssignment: ( experimentName: string ): ExperimentAssignment => {
-			if ( ! Validation.isName( experimentName ) ) {
-				throw new Error( `Invalid experimentName: ${ experimentName }` );
-			}
+			try {
+				if ( ! Validation.isName( experimentName ) ) {
+					safeLogError( {
+						message: `Invalid experimentName: ${ experimentName }`,
+						experimentName,
+						source: 'dangerouslyGetExperimentAssignment',
+					} );
+					return createFallbackExperimentAssignment( experimentName );
+				}
 
-			const storedExperimentAssignment = experimentAssignmentStore.retrieve( experimentName );
-			if ( ! storedExperimentAssignment ) {
-				throw new MissingExperimentAssignmentError(
-					`Trying to dangerously get an ExperimentAssignment that hasn't loaded.`
-				);
-			}
+				const storedExperimentAssignment = experimentAssignmentStore.retrieve( experimentName );
+				if ( ! storedExperimentAssignment ) {
+					safeLogError( {
+						message: `Trying to dangerously get an ExperimentAssignment that hasn't loaded.`,
+						experimentName,
+						source: 'dangerouslyGetExperimentAssignment',
+					} );
+					return createFallbackExperimentAssignment( experimentName );
+				}
 
-			// We want to be loud in development mode to help pick up any issues:
-			if ( config.isDevelopmentMode ) {
-				// Highlight when we dangerously get an experiment too soon to when we load one:
-				if (
-					storedExperimentAssignment &&
-					Timing.monotonicNow() - storedExperimentAssignment.retrievedTimestamp < 1000
-				) {
-					throw new Error(
-						`Warning: Trying to dangerously get an ExperimentAssignment too soon after loading it.`
-					);
+				// We want to be loud in development mode to help pick up any issues:
+				if ( config.isDevelopmentMode ) {
+					// Highlight when we dangerously get an experiment too soon to when we load one:
+					if (
+						storedExperimentAssignment &&
+						Timing.monotonicNow() - storedExperimentAssignment.retrievedTimestamp < 1000
+					) {
+						safeLogError( {
+							message: `Warning: Trying to dangerously get an ExperimentAssignment too soon after loading it.`,
+							experimentName,
+							source: 'dangerouslyGetExperimentAssignment',
+						} );
+					}
 				}
-			}
 
-			return storedExperimentAssignment;
+				return storedExperimentAssignment;
+			} catch ( error ) {
```

Comment:
Since we're catching all errors, we can keep all the `throw`s above, which would simplify things a bit.

Reply:
Yeah I was on the fence about it but why not.